dkg/src/dkg_manager/mod.rs

```diff
@@ -40,29 +95,199 @@ impl<DKG: DKGTrait, P: DKGPrivateParamsProvider<DKG>> DKGManager<DKG, P> {
             vtxn_pool_write_cli,
             agg_trx_tx: None,
             agg_trx_producer,
+            stopped: false,
+            state: InnerState::NotStarted,
         }
     }
 
     pub async fn run(
-        self,
-        _in_progress_session: Option<DKGSessionState>,
-        _dkg_start_event_rx: aptos_channel::Receiver<(), DKGStartEvent>,
-        _rpc_msg_rx: aptos_channel::Receiver<(), (AccountAddress, IncomingRpcRequest)>,
-        _dkg_txn_pulled_rx: vtxn_pool::PullNotificationReceiver,
+        mut self,
+        in_progress_session: Option<DKGSessionState>,
+        mut dkg_start_event_rx: aptos_channel::Receiver<(), DKGStartEvent>,
+        mut rpc_msg_rx: aptos_channel::Receiver<(), (AccountAddress, IncomingRpcRequest)>,
+        mut dkg_txn_pulled_rx: vtxn_pool::PullNotificationReceiver,
         close_rx: oneshot::Receiver<oneshot::Sender<()>>,
     ) {
+        let (agg_trx_tx, mut agg_trx_rx) = aptos_channel::new(QueueStyle::KLAST, 1, None);
+        self.agg_trx_tx = Some(agg_trx_tx);
+
```

Comment:
missing a `setup_deal_broadcast` call that consumes `in_progress_session` to handle crash recovery